docs/language/glossary.md

```diff
@@ -0,0 +1,314 @@
+
+This is only used for resources, as they use the move operator. [More info](https://docs.onflow.org/cadence/language/values-and-types/#force-assignment-operator--)
+
+```cadence
+pub resource R {}
+
+var a: @R? <- nil
+a <-! create R()
+```
+
+## <-> (lower than, hyphen, greater than) (Swap operator)
+
+`<->` is referred to as the Swap operator. It swaps values between the variables to the left and right of it. [More info](https://docs.onflow.org/cadence/language/operators/#swapping)
+
+```cadence
+let a = 1;  
+let b = 2;
+
+a <-> b;
+// a = 2
+// b = 1
+```
+
+## + (plus), - (minus), * (asterisk), % (percentage sign)
+
+These are all typical arithmetic operators. 
+
+- Addition: +
+- Subtraction: -
+- Multiplication: *
+- Remainder: %
+
+[More info](https://docs.onflow.org/cadence/language/operators/#arithmetic)
+
+## ? (question mark)
+
+The ? (question mark) symbol has several uses. If a ? follows a variable/constant, it represents an optional. An optional can either have a value or *nothing at all*. 
+
+```cadence
+// Declare a constant which has an optional integer type
+//
+let a: Int? = nil
+```
+
+When you see as?, that's a conditional downcasting operator. It can be used to downcast a value to a type. This operator returns an optional, and if the value has a type that is a subtype it will return the value as that type, otherwise it will return `nil`. [More info](https://docs.onflow.org/cadence/language/values-and-types/#conditional-downcasting-operator)
+
+```cadence
+// a simple interface that expects a property count
+resource interface HasCount {
+  count : Int
+}
+
+// a Counter resource that conforms to HasCount
+resource Counter: HasCount {
+      pub var count: Int
+
+    pub init(count: Int) {
+        self.count = count
+    }
+}
+
+// set a reference countRef to &counter with the hasCount interface 
+// this is important because ONLY methods in HasCount will be available!
+let countRef: &{HasCount} = &counter as &{HasCount}
+
+// BUT, we could also optionally downcast it to Counter
+let authCountRef: auth &{HasCount} = &counter as auth &{HasCount}
+let countRef2: &Counter = authCountRef as? &Counter
+```
+
+It is a big topic, so best to [read the documentation on it](https://docs.onflow.org/cadence/language/values-and-types/#optionals)
+
+
+It can also be used in ternary operations to represent the "otherwise" section, such as the following:
+
+
+```cadence
+let a = 1 > 2 ? 3 : 4
+// should be read as: 
+//   "is 1 greater than 2?"
+//   "if YES, then set a = 3,
+//   "if NO, then set a = 4.
+```
+
+It can also be used as a nil-coalescing operator. The nil-coalescing operator `??` returns the value inside an optional if it contains a value, or returns an alternative value if the optional has no value, i.e., the optional value is nil. The nil-coalescing operator can only be applied to values which have an optional type. [More info](https://docs.onflow.org/cadence/language/values-and-types/#nil-coalescing-operator)
```

Comment:
Breaking down paragraphs line-by-line helps with review and updating them in the future:

```suggestion
It can also be used as a nil-coalescing operator. 
The nil-coalescing operator `??` returns the value inside an optional if it contains a value, 
or returns an alternative value if the optional has no value, i.e., the optional value is nil. 
The nil-coalescing operator can only be applied to values which have an optional type. 
[More info](https://docs.onflow.org/cadence/language/values-and-types/#nil-coalescing-operator)
```